### PR TITLE
Add external key manager configuration docs for Choreo Connect

### PR DESCRIPTION
**Pull request by HeshanSudarshana**
## Purpose
This PR adds a page to configure external key managers and moved security based configurations to enforcer configs.

### CODE REVIEW

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/security/api-authentication/configuring-an-external-key-manager.md [hunk 1]
```diff
@@ -1,108 +1,19 @@
 # Configuring an External Key Manager
 
-WSO2 API Microgateway can be configured with an external key manager. Token introspection is
+WSO2 Choreo Connect provides the capability to configure external key managers through APIM admin portal. The issuer data is retrieved from the event hub at the startup and updated as the changes are made from the admin portal.
```
Comment (menakaj) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/security/api-authentication/configuring-an-external-key-manager.md:3]:
```suggestion
Choreo Connect provides the capability to configure external key managers through APIM admin portal. The issuer data is retrieved from the event hub at the startup and updated as the changes are made from the admin portal.
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/security/api-authentication/configuring-an-external-key-manager.md [hunk 2]
```diff
@@ -1,108 +1,19 @@
...
 
-[OAuth 2.0 Token Introspection](https://tools.ietf.org/html/rfc7662) is a specification that defines how a protected resource can validate and obtain meta-information of OAuth2.0 tokens issued by an authorization server. This specification defines an introspect endpoint that provides meta-information including the token validity of a given OAuth2.0 token.  The introspection endpoint is implemented on the authorization server-side.
+### Step 1 - Configure Choreo Connect with API Manager
 
-WSO2 Microgateway can be configured to connect with an introspect endpoint (/introspect) of an external key manager.  If the API Microgateway receives an opaque (non-JWT) token, then the external key manager's introspect endpoint will be called in order to validate the token upon configuration is enabled. Microgateway will cache the tokens once it is identfied as an valid token, upon recieving the same token it will take the decision from the cache without connecting again with the key manager.
+Please refer [here]({{base_path}}/deploy-and-publish/deploy-on-gateway/choreo-connect/concepts/apim-as-control-plane/#configure-choreo-connect-with-api-manager).
```
Comment (menakaj) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/security/api-authentication/configuring-an-external-key-manager.md:7]:
```suggestion
Please refer [configuring Choreo Connect with API Manager]({{base_path}}/deploy-and-publish/deploy-on-gateway/choreo-connect/concepts/apim-as-control-plane/#configure-choreo-connect-with-api-manager).
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/security/api-authentication/configuring-an-external-key-manager.md [hunk 3]
```diff
@@ -1,108 +1,19 @@
...
 
-### Enable introspection with external key managers
+### Step 2 - Add the external key manager to API Manager
 
-Microgateway by default connect with WSO2 API Manager key validation service which is different from standard introspection. Provide following configurations in the `micro-gw.conf` to configure key manager. In order to enable introspection with external key manager "external" parameter should be set under the key manager configurations.
+Please refer [here]({{base_path}}/administer/key-managers/overview) to configure the desired key manager.
```
Comment (menakaj) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/security/api-authentication/configuring-an-external-key-manager.md:11]:
```suggestion
Please refer [Multiple Key Manager Support in WSO2 API Manager]({{base_path}}/administer/key-managers/overview) to configure the desired key manager. 
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/configurations/enforcer-configurations.md [hunk 4]
```diff
@@ -10,6 +10,9 @@ The following are the configurations with regard to Enforcer. The configuration
...
+|`security.authHeader` | The configurations required for enforcer to provide API authorization security for Choreo Connect and the backend. By default `enableOutboundAuthHeader` is false which removes the authorization header from the backend request.  `authorizationHeader` configuration defines the authorization header expected by the Choreo Connect. It can be overridden at API level using the `x-auth-header` extension. |
+|`security.tokenService` | defines the configuration required to JWT token authorization. You can provide multiple JWT issuers. When the Choreo Connect connects with WSO2 API Manager, the tokenService configurations defined at `config.toml` are overridden by the Key Manager configurations received from API Manager if the same issuer persists in both sides and store at issuer data store. Whenever a Key Manager is deleted by API Manager Admin Portal, it is not getting removed from issuer data store, if it is configured as a token service. For more information refer, [Configuring and external key manager]({{base_path}}/deploy-and-publish/deploy-on-gateway/choreo-connect/security/api-authentication/configuring-an-external-key-manager) |
```
Comment (menakaj) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/configurations/enforcer-configurations.md:15]:
```suggestion
|`security.tokenService` | Defines the configuration required to JWT token authorization. You can provide multiple JWT issuers. When the Choreo Connect connects with WSO2 API Manager, the tokenService configurations defined at `config.toml` are overridden by the Key Manager configurations received from API Manager if the same issuer persists in both sides and store at issuer data store. Whenever a Key Manager is deleted by API Manager Admin Portal, it is not getting removed from issuer data store, if it is configured as a token service. For more information refer, [Configuring and external key manager]({{base_path}}/deploy-and-publish/deploy-on-gateway/choreo-connect/security/api-authentication/configuring-an-external-key-manager) |
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/security/api-authentication/configuring-an-external-key-manager.md (hunk 1 above)
Comment (HeshanSudarshana) [reply en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/security/api-authentication/configuring-an-external-key-manager.md]: Added with https://github.com/wso2/docs-apim/pull/3769/commits/858d270db004a2d3a629254b6f85eb12be424c17

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/security/api-authentication/configuring-an-external-key-manager.md (hunk 2 above)
Comment (HeshanSudarshana) [reply en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/security/api-authentication/configuring-an-external-key-manager.md]: Added with https://github.com/wso2/docs-apim/pull/3769/commits/858d270db004a2d3a629254b6f85eb12be424c17

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/security/api-authentication/configuring-an-external-key-manager.md (hunk 3 above)
Comment (HeshanSudarshana) [reply en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/security/api-authentication/configuring-an-external-key-manager.md]: Added with https://github.com/wso2/docs-apim/pull/3769/commits/858d270db004a2d3a629254b6f85eb12be424c17

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/configurations/enforcer-configurations.md (hunk 4 above)
Comment (HeshanSudarshana) [reply en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/configurations/enforcer-configurations.md]: Added with https://github.com/wso2/docs-apim/pull/3769/commits/858d270db004a2d3a629254b6f85eb12be424c17

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-api-to-choreo-connect.md [hunk 5]
```diff
@@ -12,61 +12,15 @@ There are two ways to add an API to the Choreo Connect.
...
 
-### Step 4 - Create an API in API Manager
+### Step 2 - Create an API in API Manager
 
 Follow the steps [here]({{base_path}}/design/create-api/create-rest-api/create-a-rest-api/).
 
-### Step 5 - Deploy the API in API Manager
+### Step 3 - Deploy the API in API Manager
 
  The guide [here]({{base_path}}/deploy-and-publish/deploy-on-gateway/deploy-api/deploy-an-api) will explain how you could easily deploy the API you just created.
```
Comment (menakaj) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-api-to-choreo-connect.md:25]: Shall we fix these as well? Replace [here] with proper topics. Fix in other places as well.

**Review by Rajith90 — APPROVED** [review]

**Pull request merged**